tests/test_arrow_dataset.py

```diff
@@ -2365,6 +2382,39 @@ def test_to_pandas(self, in_memory):
                     for col_name in dset.column_names:
                         self.assertEqual(len(dset_to_pandas[col_name]), dset.num_rows)
 
+    @require_polars
+    def test_to_polars(self, in_memory):
+        import polars as pl
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            # Batched
+            with self._create_dummy_dataset(in_memory, tmp_dir, multiple_columns=True) as dset:
+                batch_size = dset.num_rows - 1
+                to_polars_generator = dset.to_polars(batched=True, batch_size=batch_size)
+
+                for batch in to_polars_generator:
+                    self.assertIsInstance(batch, pl.DataFrame)
+                    self.assertListEqual(sorted(batch.columns), sorted(dset.column_names))
+                    for col_name in dset.column_names:
+                        self.assertLessEqual(len(batch[col_name]), batch_size)
+
+                # Full
+                dset_to_polars = dset.to_polars()
+                self.assertIsInstance(dset_to_polars, pl.DataFrame)
+                self.assertListEqual(sorted(dset_to_polars.columns), sorted(dset.column_names))
+                for col_name in dset.column_names:
+                    self.assertEqual(len(dset_to_polars[col_name]), len(dset))
+
+                # With index mapping
+                with dset.select([1, 0, 3]) as dset:
+                    dset_to_polars = dset.to_polars()
+                    self.assertIsInstance(dset_to_polars, pl.DataFrame)
+                    self.assertEqual(len(dset_to_polars), 3)
+                    self.assertListEqual(sorted(dset_to_polars.columns), sorted(dset.column_names))
+
+                    for col_name in dset.column_names:
+                        self.assertEqual(len(dset_to_polars[col_name]), dset.num_rows)
```

Comment:
Before closing the temporary directory it's important to delete all the references to opened files on Windows

Can you try adding this ? Hopefully this fixes the Windows CI
```suggestion
                        self.assertEqual(len(dset_to_polars[col_name]), dset.num_rows)
                    del dset_to_polars
```